### PR TITLE
refactor code to flutter 1.12.x

### LINKED ISSUE #300: Refactor code to Flutter version 1.12.13

### PR DESCRIPTION
**Pull request by mathiasandresen**
closing #300 
* Fixes unit tests
* Refactors the code to follow the flutter linter (`flutter analyze`)

### CODE REVIEW

File: test/widgets/giraf_button_widget_test.dart [hunk 1]
```diff
@@ -5,6 +5,9 @@ import 'package:rxdart/rxdart.dart';
 import 'package:weekplanner/widgets/giraf_button_widget.dart';
 import 'package:flutter/material.dart';
 
+
+const ImageIcon myIcon = ImageIcon(AssetImage('assets/icons/accept.png'));
```
Comment (Jakobzs) [inline test/widgets/giraf_button_widget_test.dart]: Maybe “acceptIcon”?

File: lib/widgets/giraf_button_widget.dart [hunk 2]
```diff
@@ -158,7 +158,7 @@ class _GirafButtonState extends State<GirafButton> {
   Widget _buildWidgetsOnButton() {
     const TextStyle textStyle = TextStyle(color: Colors.black, fontSize: 20);
 
-    if (widget.text != null && widget.icon != null)
+    if (widget.text != null && widget.icon != null) {
```
Comment (Jakobzs) [inline lib/widgets/giraf_button_widget.dart:161]: Is it necessary to use { and } when only a single statement follows? (Either way works, just curious)

File: test/widgets/giraf_button_widget_test.dart (hunk 1 above)
Comment (mads-bo) [reply test/widgets/giraf_button_widget_test.dart]: Changed in 3a2f8b8

File: lib/widgets/giraf_button_widget.dart (hunk 2 above)
Comment (mads-bo) [reply lib/widgets/giraf_button_widget.dart:161]: It's enforced by the linter

**Review by Jakobzs — APPROVED** [review]

**Review by MrPekar98 — APPROVED** [review]

**Review by rasmus91 — APPROVED** [review]

**Pull request merged**